dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go

```diff
@@ -291,7 +293,13 @@ func generateTarball(hadoopClients []string) error {
 	}
 
 	mvnArgs := getCommonMvnArgs(hadoopVersion)
-	run("compiling repo", "mvn", mvnArgs...)
+	if skipUIFlag {
+		mvnArgsNoUI := append(mvnArgs, "-pl", "!webui")
+		run("compiling repo without UI", "mvn", mvnArgsNoUI...)
```

Comment:
is it really necessary to add this to the output string? Just seems bad to have the same command with a different string. If you really want it then,

```golang
var output
if skipUIFlag {
  output = "compiling repo without UI"
else {
  output = "compiling repo"
}
run(output, mvnArgs...)
```

Reply:
The mvnArgs are different in each case, so it's not just printing a different output. There's a second string array mvnArgsNoUI which is passed if skip-ui is set.

Reply:
I think this would be better then?

```golang
var output
if skipUIFlag {
  mvnArgs = append(mvnArgs, "-pl", "!webui")
  output = "compiling repo without UI"
else {
  output = "compiling repo"
}
run(output, mvnArgs...)
```

Reply:
The issue is if you append the -pl !webui to the mvnArgs, then all the UFS modules attempt to build with those arguments as well. Since webui doesn't exist in the underfs module, it throws a reactor error. This was a clean way for me to only skip the webui when building the parent project without affecting the mvnArgs for subsequent modules.

Reply:
I see. I didn't realize that the mvnArgs object was used down below